beets/dbcore/query.py

```diff
@@ -428,6 +416,28 @@ def col_clause(self) -> Tuple[str, Sequence[SQLiteType]]:
                 return "1", ()
 
 
+class InQuery(FieldQuery[Sequence[AnySQLiteType]]):
+    """Query which matches values in the given set."""
+
+    field: str
+    pattern: Sequence[AnySQLiteType]
+    fast: bool = True
+
+    @property
+    def subvals(self) -> Sequence[AnySQLiteType]:
```

Comment:
Why do we need `subvals`? Couldn't subclass constructors simply convert the `pattern` to the appropriate type and pass that to `InQuery.__init__`?

Reply:
Ah I didn't end up adding a docstring here - but yes you got the purpose of this right.

I saw two reasons for this:

1. Conversion to `memoryview` is only required because we use SQLite backend, and since this class has wider responsibilities than simply querying the database, it didn't seem right to couple its implementation to what's required by SQLite. This mostly comes from the Dependency inversion principle.

    Relatedly, storing items as a different type may break this method

    ```python
    @classmethod
    def value_match(
        cls, pattern: Sequence[AnySQLiteType], value: AnySQLiteType
    ) -> bool:
        return value in pattern
    ```

2. Regarding the implementation, in my head `__init__` is responsible for initialising the object, i.e. resolving the attributes that it needs, in this case the list of filenames to be stored as `pattern`. Given what I mentioned above, I see type conversion to `memoryview` as a separate task that only needs to happen when we're querying the database. Thus I moved it away from `__init__` to a separate method/attribute that has only this responsibility. I should have named it appropriately though!

Reply:
Ok, sounds good! I don't think it's absolutely necessary to have a docstring here (that'd be way more strict than what we have in many other places).